### PR TITLE
created serve-static middleware for node

### PR DESCRIPTION
**Pull request by that-ambuj**
It is the serve-static implementation as requested in issue #3. there are no major changes. I have exported both serve and serveStatic from index.ts.

### CODE REVIEW

File: src/serve-static.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { existsSync, readFileSync } from 'fs'
+import type { Handler } from 'hono/dist/hono'
```
Comment (yusukebe) [inline src/serve-static.ts:2]: fix path `hono/dist/hono to `hono`

**Review by yusukebe — CHANGES_REQUESTED** [review]

**Comment by yusukebe** [comment]
And, please write the tests for Serve Static middleware. If there is no test, we will not accept the PR.

**Comment by that-ambuj** [comment]
I have written the tests and fixed the import from hono. Please review the PR. Thanks!

**Comment by that-ambuj** [comment]
Also should I update honojs.dev for this middleware?

**Comment by yusukebe** [comment]
Hi @that-ambuj !

Nice! I've commented on some points, please fix these.

> Also should I update honojs.dev for this middleware?

You don't have to do it now.

File: test/serve-static.test.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  const server = createAdaptorServer(app)
+
+  it('Should return index.html', async () => {
+    const res = await request(server).get('/static/')
+    expect(res.status).toBe(200)
+    expect(res.text).toBe('<h1>Hello Hono</h1>')
+    expect(res.headers['content-type']).toBe('text/html; charset=utf-8')
+  })
+
+  it('Should return hono.html', async () => {
```
[the remark this answers is not in the document]
Comment (that-ambuj) [reply test/serve-static.test.ts:36]: It is just for testing basic routing. You know, how you would go from one html file to another using a link.

File: test/serve-static.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import { createAdaptorServer } from './../src/server'
+import { Hono } from 'hono'
+import request from 'supertest'
+import { serveStatic } from './../src/'
+
+// Mock
```
[the remark this answers is not in the document]
Comment (that-ambuj) [reply test/serve-static.test.ts:6]: I was just thinking of stealing the tests from you. :)

[the remark this answers is not in the document]
Comment (yusukebe) [reply test/serve-static.test.ts:6]: These are for Cloudflare Workers, so these are not needed.

File: test/serve-static.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (that-ambuj) [reply test/serve-static.test.ts:36]: Should I still omit this test?

[the remark this answers is not in the document]
Comment (yusukebe) [reply test/serve-static.test.ts:36]: Yes, I know. Let's leave it as it is here.

**Comment by that-ambuj** [comment]
Changes have been made!

File: test/assets/static/data.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+    "id": 1,
+    "name": "Foo Bar",
+    "isBot": true
```
Comment (yusukebe) [inline test/assets/static/data.json:4]: I'm not sure what `isBot` means, so how about `flag`?

File: test/serve-static.test.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    expect(res.status).toBe(200)
+    expect(res.body).toEqual({
+      id: '1',
+      name: 'Foo Bar',
+      isBot: 'true',
+    })
+    expect(res.headers['content-type']).toBe('application/json; charset=utf-8')
+  })
+
+  it('Should return correct headers and data for text', async () => {
```
[the remark this answers is not in the document]
Comment (yusukebe) [reply test/serve-static.test.ts:45]: Shall we delete this test?

**Review by yusukebe — CHANGES_REQUESTED** [review]

**Comment by yusukebe** [comment]
Two reviews are still pending.

File: test/assets/static/data.json (hunk 4 above)
Comment (that-ambuj) [reply test/assets/static/data.json:4]: I'm sorry, I didn't notice.

File: test/serve-static.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (that-ambuj) [reply test/serve-static.test.ts:45]: It is in tests for bun and cloudflare. So, I think we should keep it for consistency.

**Comment by that-ambuj** [comment]
Should I make more changes?

**Review by yusukebe — APPROVED** [review]

**Comment by yusukebe** [comment]
@that-ambuj 

Nothing! All done!
Thank you for all your hard work for a long time. I will merge it now.

**Comment by that-ambuj** [comment]
Thanks a lot for letting me work on this!

**Pull request merged**